wasmtime-wasi/src/host_impls.rs

```diff
@@ -1,5 +1,148 @@
+    extern "C" {
+        #[thread_local]
+        static errno: c_int;
+    }
+
+    unsafe { errno as i32 }
+}
+
+/// Convert POSIX error code to host's WASI error code
+fn convert_errno(errno: i32) -> host::__wasi_errno_t {
+    #[allow(unreachable_patterns)]
+    match errno {
+        libc::E2BIG => host::__WASI_E2BIG,
+        libc::EACCES => host::__WASI_EACCES,
+        libc::EADDRINUSE => host::__WASI_EADDRINUSE,
+        libc::EADDRNOTAVAIL => host::__WASI_EADDRNOTAVAIL,
+        libc::EAFNOSUPPORT => host::__WASI_EAFNOSUPPORT,
+        libc::EAGAIN | libc::EWOULDBLOCK => host::__WASI_EAGAIN,
+        libc::EALREADY => host::__WASI_EALREADY,
+        libc::EBADF => host::__WASI_EBADF,
+        libc::EBADMSG => host::__WASI_EBADMSG,
+        libc::EBUSY => host::__WASI_EBUSY,
+        libc::ECANCELED => host::__WASI_ECANCELED,
+        libc::ECHILD => host::__WASI_ECHILD,
+        libc::ECONNABORTED => host::__WASI_ECONNABORTED,
+        libc::ECONNREFUSED => host::__WASI_ECONNREFUSED,
+        libc::ECONNRESET => host::__WASI_ECONNRESET,
+        libc::EDEADLK => host::__WASI_EDEADLK,
+        libc::EDESTADDRREQ => host::__WASI_EDESTADDRREQ,
+        libc::EDOM => host::__WASI_EDOM,
+        libc::EDQUOT => host::__WASI_EDQUOT,
+        libc::EEXIST => host::__WASI_EEXIST,
+        libc::EFAULT => host::__WASI_EFAULT,
+        libc::EFBIG => host::__WASI_EFBIG,
+        libc::EHOSTUNREACH => host::__WASI_EHOSTUNREACH,
+        libc::EIDRM => host::__WASI_EIDRM,
+        libc::EILSEQ => host::__WASI_EILSEQ,
+        libc::EINPROGRESS => host::__WASI_EINPROGRESS,
+        libc::EINTR => host::__WASI_EINTR,
+        libc::EINVAL => host::__WASI_EINVAL,
+        libc::EIO => host::__WASI_EIO,
+        libc::EISCONN => host::__WASI_EISCONN,
+        libc::EISDIR => host::__WASI_EISDIR,
+        libc::ELOOP => host::__WASI_ELOOP,
+        libc::EMFILE => host::__WASI_EMFILE,
+        libc::EMLINK => host::__WASI_EMLINK,
+        libc::EMSGSIZE => host::__WASI_EMSGSIZE,
+        libc::EMULTIHOP => host::__WASI_EMULTIHOP,
+        libc::ENAMETOOLONG => host::__WASI_ENAMETOOLONG,
+        libc::ENETDOWN => host::__WASI_ENETDOWN,
+        libc::ENETRESET => host::__WASI_ENETRESET,
+        libc::ENETUNREACH => host::__WASI_ENETUNREACH,
+        libc::ENFILE => host::__WASI_ENFILE,
+        libc::ENOBUFS => host::__WASI_ENOBUFS,
+        libc::ENODEV => host::__WASI_ENODEV,
+        libc::ENOENT => host::__WASI_ENOENT,
+        libc::ENOEXEC => host::__WASI_ENOEXEC,
+        libc::ENOLCK => host::__WASI_ENOLCK,
+        libc::ENOLINK => host::__WASI_ENOLINK,
+        libc::ENOMEM => host::__WASI_ENOMEM,
+        libc::ENOMSG => host::__WASI_ENOMSG,
+        libc::ENOPROTOOPT => host::__WASI_ENOPROTOOPT,
+        libc::ENOSPC => host::__WASI_ENOSPC,
+        libc::ENOSYS => host::__WASI_ENOSYS,
+        // TODO: verify if this is correct
+        #[cfg(target_os = "freebsd")]
+        libc::ENOTCAPABLE => host::__WASI_ENOTCAPABLE,
```

Comment:
Yes, this is correct, at least at present.

Reply:
:+1: